checkpointing/tensorflow/README.md

```diff
@@ -0,0 +1,21 @@
+## Checkpointing in Tensorflow with conda environment and GPU support
+The example uses a pre-configured conda environment to demonstrate how one might implement model checkpointing in Tensorflow.
+
+The Tensorflow example downloads the MNIST dataset and trains an neural net on it for 20 epochs, making a "checkpoint" every fifth epoch. In the event of a runtime interruption, these checkpoints are returned to the user and can be used to resume training, preventing the loss of already-completed work.
+
+The five files include:
+- `checkpoint.sub` ## used to submit the job on HTCondor
+- `run.sh` ## the executable called by `checkpoint.sub`
+- `tf_checkpointing.py` ## the python program with an implementation of checkpointing
+- `tf_checkpointing.tar.gz` ## the conda environment used by HTCondor for tensorflow dependencies -- this is hosted on a squid server and is not included in the repo.
+- `environment.yml` ## An environment file for creating the conda env used in this example
+
+### Usage
+- log into the HTC system.
+- clone this repository: `git clone https://github.com/CHTC/template-GPUs`
+- `cd` into this folder: `cd template-GPUs/checkpointing/conda/tensorflow`
+- because conda environments tend to be large, a Squid caching server is used to host the environment. You can access the environment .tar.gz file from the submit node at /squid/gpu-examples. For more information about using Squid, please review the CHTC guide:
+[Large File Availability Via Squid](https://chtc.cs.wisc.edu/uw-research-computing/file-avail-squid) guide.
+- submit the sample job: `condor_submit checkpoint.sub`
+- upon completion, HTCondor will return a zipped model file, `model.tar.gz`, along with checkpoint files `checkpoint.h5` (checkpointed model) and `checkpoint.txt` (which epoch to resume training on)
+- in the event of a runtime error, HTCondor will return only `checkpoint.h5` and `checkpoint.txt`, assuming it has reached at least the first checkpoint
```

Comment:
Do these files actually get returned to the home directory? I think it's important to be clear that checkpoint files actually go to the "`SPOOL`" directory - we can provide more breadcrumbs on how to find them if we think that's important. Otherwise I would change "return" to "preserve".

Reply:
In this implementation, those files actually do ship back to the submit server at the end of the job. If we want to change it so they don't (I think just sticking them in a folder will prevent this, won't it?) I can do that.

Reply:
ah, I think I misunderstood. is that last bullet point if the job *fails* not if it's just interrupted?

Reply:
I think I should change the last bullet to highlight that the files transfer to the spool, which is how condor is able to resume the job in the event of an interruption.

Reply:
yes, that would be more accurate.